packages/validator/src/services/block.ts

```diff
@@ -80,7 +80,9 @@ export class BlockProposingService {
         this.metrics?.blockProposingErrors.inc({error: "produce"});
         throw extendError(e, "Failed to produce block");
       });
-      this.logger.debug("Produced block", debugLogCtx);
+      const blockFeeRecipient = (block.data as bellatrix.BeaconBlock).body.executionPayload?.feeRecipient;
+      const feeRecipient = blockFeeRecipient !== undefined ? toHexString(blockFeeRecipient) : undefined;
+      this.logger.debug("Produced block", {...debugLogCtx, feeRecipient});
```

Comment:
I suppose validator should confirm the `feeReceipient` to address this in this beacon node api spec: `Note that there is no guarantee that the beacon node will use the supplied fee recipient when creating a block proposal, so on receipt of a proposed block the validator should confirm that it finds the fee recipient within the block acceptable before signing it.`

an attacker can periodically poll our api periodically to provide its own addresses

Reply:
yes correct observation :+1: , one thing to consider is that the block build by mev-boost might not have the payout to `feeRecipient` as block's `feeRecipient` but in form of transaction included last minute. However we can address that in the mev's PR :thinking:

Reply:
just have a look at `ProduceBlindedBlockResponse`, I suppose we'll still have `feeRecipient` in `BlindedBlock` version, we'll only have transaction root instead of transaction list (correct me if I'm wrong, just follow (https://ethereum.github.io/beacon-APIs/#/Validator/produceBlindedBlock)

Reply:
yes ofcourse, what i meant was that the mev builders can put their own fee Recipient and instead reward the fee Recipient by including a transaction in block. Will confirm with other teams how they are handling such scenario, but we can address that in MEV PR. right now, i will introduce the check.